### PR TITLE
Extend CRD: allow to define storageInitializerImage in the graph definition

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

This allows to define image used by the `Storage Initializer` as part of the `PredictiveUnit` definition - `graph` section in the `CR` e.g.
```yaml
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  name: rclone-sklearn
spec:
  predictors:
  - name: default
    replicas: 1
    graph:
      name: classifier
      implementation: SKLEARN_SERVER
      modelUri: s3://sklearn/iris
      storageInitializerImage: gcr.io/kfserving/storage-initializer:v0.4.0
      envSecretRefName: seldon-init-container-secret
```

This allow to have elegant way to overwrite used `storageInitializerImage` on per-deployment basis.

**Which issue(s) this PR fixes**:

Fixes https://github.com/SeldonIO/seldon-core/issues/2821

**Special notes for your reviewer**:

To be finished:
- [x] unit test
- [x] doc entry and example

**Does this PR introduce a user-facing change?**:

```release-note
One can now define image used by `storageInitializer` init container by defining `storageInitializerImage` in the `graph` section of `SeldonDeployment` CR definition. 
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Feb  9 10:39:22 UTC 2021
The logs for [integration] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2937/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2937 --build=2

**Comment by seldondev** [comment]
Tue Feb  9 10:39:23 UTC 2021
The logs for [notebooks] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2937/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2937 --build=1

**Comment by RafalSkolasinski** [comment]
This should be ready for review now.

**Comment by seldondev** [comment]
Tue Feb  9 18:15:15 UTC 2021
The logs for [integration] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2937/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2937 --build=4

**Comment by seldondev** [comment]
Tue Feb  9 18:15:16 UTC 2021
The logs for [notebooks] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2937/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2937 --build=3

**Comment by seldondev** [comment]
Thu Feb 11 11:41:24 UTC 2021
The logs for [integration] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2937/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2937 --build=6

**Comment by seldondev** [comment]
Thu Feb 11 11:41:28 UTC 2021
The logs for [notebooks] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2937/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2937 --build=5

**Comment by cliveseldon** [comment]
Need to rebase and rerun tests

File: examples/init_containers/Makefile
Comment (cliveseldon) [inline examples/init_containers/Makefile:3]: Why is image version not based on release. Is it assumed it is unlikely to change be effected by release. If so maybe remove the commented line?

File: operator/Makefile
Comment (cliveseldon) [inline operator/Makefile:58]: install all doesn't make sense to me - you either need to install pre 1.18 CRD or post 1.18 CRD

File: operator/controllers/seldondeployment_prepackaged_servers.go
Comment (cliveseldon) [inline operator/controllers/seldondeployment_prepackaged_servers.go:136]: why do we need a separate function for this as it just returns the value from `pu`

**Comment by seldondev** [comment]
failed to trigger Pull Request pipeline
* failed to create agent
* failed to calculate in repo config
* failed to load trigger config for repository SeldonIO/seldon-core for ref e9cde00daf7a5bb9d3b7c122c354c02096ef3212
* failed to find any lighthouse configuration files in repo SeldonIO/seldon-core at sha e9cde00daf7a5bb9d3b7c122c354c02096ef3212
* failed to process repo SeldonIO/seldon-core refref e9cde00daf7a5bb9d3b7c122c354c02096ef3212
* failed to list files in directory /var/tmp/gitrepo598506571/.lighthouse
* open /var/tmp/gitrepo598506571/.lighthouse
* no such file or directory

**Review by cliveseldon — CHANGES_REQUESTED** [review]
a few small changes

File: examples/init_containers/Makefile
Comment (RafalSkolasinski) [reply examples/init_containers/Makefile:3]: This is only an example that actually has no dependency on the in-repo code. It just changes the `entrypoint` of the upstream image:
```
FROM rclone/rclone:latest
ENTRYPOINT ["rclone", "copy"]
```

When later we will add a proper rclone-based alternative to storage.py (not just an example of custom init container) then I guess it will make sense to version it with `../../version.txt`.

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | aa4960752591242bbc25f66beb1564b2d49db775 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-2wvjg) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | aa4960752591242bbc25f66beb1564b2d49db775 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/notebooks-tqzhn) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[cliveseldon](https://github.com/SeldonIO/seldon-core/pull/2937#issuecomment-785791181 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
